src/serializer/ResidualFunctions.js

```diff
@@ -240,6 +243,7 @@ export class ResidualFunctions {
         body = getFunctionBody(instance);
       }
       body.push(t.variableDeclaration("var", [t.variableDeclarator(funcId, funcNode)]));
+      this.generatedFunctionIds.add(funcId);
```

Comment:
The order in which `defineFunction` is called on function is not necessarily the order in which the function declarations will appear in the code. (That's why the outer function is called `spliceFunctions`...)